alloydb-jdbc-connector/src/main/java/com/google/cloud/alloydb/InternalConnectorRegistry.java

```diff
@@ -95,19 +97,29 @@ public void close(String name) {
   }
 
   /** Shutdown all connectors and remove the singleton instance. */
-  public void shutdown() {
+  private void shutdown() {
     this.unnamedConnectors.forEach((key, c) -> c.close());
     this.unnamedConnectors.clear();
     this.namedConnectors.forEach((key, c) -> c.close());
     this.namedConnectors.clear();
     this.executor.shutdown();
   }
 
+  private void resetExecutor() {
+    this.executor = MoreExecutors.listeningDecorator(Executors.newScheduledThreadPool(pool_size));
+  }
+
   @Override
   public void close() {
     shutdown();
   }
 
+  /** Calls shutdown on the singleton and reset the executor. */
+  public void resetInstance() {
+    shutdown();
```

Comment:
Could we update this method to close all open connectors and otherwise leave the executor running?